stores/healthcareProfessionalsStore.ts

```diff
@@ -15,6 +20,61 @@ export const useHealthcareProfessionalsStore = defineStore(
         const healthcareProfessionalsData: Ref<HealthcareProfessional[]>
         = ref([])
         const selectedHealthcareProfessionalId: Ref<string> = ref('')
+        const selectedHealthcareProfessionalData: Ref<HealthcareProfessional | undefined> = ref()
+        const healthcareProfessionalSectionFields = reactive({
+            healthcareProfessionalNameArray: [] as Array<LocalizedNameInput>,
+            localizedFirstName: '' as string,
+            localizedLastName: '' as string,
+            localizedMiddleName: '' as string,
+            nameLocale: Locale.EnUs as Locale,
+            healthcareProfessionalAcceptedInsurances: [] as Array<Insurance>,
+            healthcareProfessionalDegrees: [] as Array<Degree>,
+            healthcareProfessionalSpecialties: [] as Array<Specialty>,
+            healthcareProfessionalLocales: [] as Array<Locale>
+        })
+
+        function setSelectedHealthcareProfessional(healthcareProfessionalId: string) {
+            selectedHealthcareProfessionalData.value = healthcareProfessionalsData.value
+                .find((healthcareProfessional: HealthcareProfessional) => healthcareProfessional.id === healthcareProfessionalId)
+        }
+
+        function initializeHealthcareProfessionalValues(selectedHealthcareProfessionalData: HealthcareProfessional | undefined) {
+            if (!selectedHealthcareProfessionalData) return
+
+            healthcareProfessionalSectionFields.healthcareProfessionalNameArray = selectedHealthcareProfessionalData.names
```

Comment:
This function seems more like a function for mapping from api data to component format rather than an initializer function. Perhaps this just needs a rename?

Also, it looks like where this is being used, we could just call this from the `setSelectedHealthcareProfessional` method internally instead of from the component. Is there a use case outside of the set method?

Reply:
I'm presuming the `setSelectedHealthcareProfessional` is potentially going to be used elsewhere without needing to initialize values. Is that right @NabbeunNabi ?